node_cli/tasks.js

```diff
@@ -0,0 +1,129 @@
+  index.output(JSON.stringify(tasks, null, 3), "result");
+};
+
+exports.createTask = async (user) => {
+  try {
+    index.output("*** CREATE NEW TASK ***", "header");
+    const task = await inquirer.prompt([
+      {
+        type: "input",
+        name: "name",
+        message: "What is the task text?",
+      },
+      {
+        type: "rawlist",
+        name: "status",
+        message: "What is the task status?",
+        choices: ["Open", "In Progress", "Closed"],
+        default: function () {
+          return "Open";
+        },
+      },
+    ]);
+
+    const result = await realmTasks.createTask(user, task.name, task.status);
+
+    index.output("New task created", "header");
+    index.output(JSON.stringify(result, " ", 3), "result");
+  } catch (err) {
+    index.output(err, "error");
+  }
+};
+
+exports.getTask = async (user) => {
+  try {
+    const task = await inquirer.prompt([
+      {
+        type: "input",
+        name: "id",
+        message: "What is the task ID (_id)?",
+      },
+    ]);
+
+    const result = await realmTasks.getTask(user, task.id);
+    if (result !== undefined) {
+      index.output("Here is the task you requested:", "header");
+      index.output(JSON.stringify(result, " ", 3), "result");
+    }
+  } catch (err) {
+    index.output(err, "error");
+  }
+};
+
+exports.deleteTask = async (user) => {
+  index.output("DELETE A TASK", "header");
+  const answers = await inquirer.prompt([
+    {
+      type: "input",
+      name: "id",
+      message: "What is the task ID (_id)?",
+    },
+    {
+      type: "confirm",
+      name: "confirm",
+      message: "Are you sure you want to delete this task?",
+    },
+  ]);
+
+  if (answers.confirm) {
+    let deleteResult = await realmTasks.deleteTask(user, answers.id);
+    if (deleteResult) index.output("Task deleted.", "result");
+    return;
+  }
+};
+
+exports.editTask = async (user) => {
+  index.output("CHANGE A TASK", "header");
+  let answers = await inquirer.prompt([
+    {
+      type: "input",
+      name: "id",
+      message: "What is the task ID (_id)?",
+    },
+    {
+      type: "input",
+      name: "key",
+      message: "What is the field you want to change?",
+    },
+    {
+      type: "input",
+      name: "value",
+      message: "What is the new value?",
+    },
+  ]);
+
+  let changeResult = await realmTasks.changeTask(user, answers);
+  index.output("Task updated.", "result");
+  index.output(changeResult, "result");
+  return;
+};
+
+exports.changeStatus = async (user) => {
+  index.output("Update Task Status", "header");
+  const answers = await inquirer.prompt([
+    {
+      type: "input",
+      name: "id",
+      message: "What is the task ID (_id)?",
+    },
+    {
+      type: "rawlist",
+      name: "value",
+      message: "What is the new status?",
+      choices: ["Open", "In Progress", "Closed"],
+    },
+  ]);
+
+  answers.key = "status";
+  let changeResult = await realmTasks.changeTask(user, answers);
```

Comment:
I think this could be simplified if you get rid of the realm-wrapping "realmTasks" object and just use the realm directly in this file.

Reply:
I could do that, but SRP: architecturally, I like having all the realm-specific stuff isolated from anything UI-related. The tasks.js file handles the prompting and user input, while realmTasks handles the calls to the realm APIs.

Reply:
Realm is the data layer. Based on discussions with iOS Realm eng they have tried to discourage people from wrapping it because it defeats the purpose of the live objects.

Reply:
I see the appeal of separate modules, but they do seem somewhat antithetical to the original design goal of Realm, i.e. "Realm Objects are just objects (with superpowers)"

Reply:
This feels true for the other wrapper files as well. `realmAuth.js` exports `login()` and `logout()` (and probably should also export `currentUser()`), which are all thin wrappers around `app` methods/properties with the same names. Why not just import the `app` object directly?

This feels particularly true given my [error handling comment](https://github.com/mongodb-university/realm-tutorial/pull/45/files#r456666029) in `users.js` - it seems like there's some extra indirection here. It may have architectural value but I think it also adds a fair amount of unnecessary complexity for learners.